File: ilias-connector/ilias-connector-api/src/main/java/com/github/thetric/iliasdownloader/service/model/IliasItem.java
package com.github.thetric.iliasdownloader.service.model;

/**
 * @author broj
 * @since 18.08.2016
 */
public interface IliasItem {
    int getId();

    String getName();

    String getUrl();

    void setId(int id);

    void setName(String name);

    void setUrl(String url);
}

File: src/main/kotlin/com/github/thetric/iliasdownloader/connector/model/IliasItem.kt
package com.github.thetric.iliasdownloader.connector.model

/**
 * Base interface for all Ilias items.
 *
 * @see Course
 *
 * @see CourseFile
 *
 * @see CourseFolder
 */
interface IliasItem {
    val name: String
    val url: String
}
